modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -128,14 +128,16 @@ static GDScriptParser::DataType make_script_meta_type(const Ref<Script> &p_scrip
 
 // In enum types, native_type is used to store the class (native or otherwise) that the enum belongs to.
 // This disambiguates between similarly named enums in base classes or outer classes
-static GDScriptParser::DataType make_enum_type(const StringName &p_enum_name, const String &p_base_name, const bool p_meta = false) {
+static GDScriptParser::DataType make_enum_type(const StringName &p_enum_name, const String &p_base_name, const bool p_meta = false, GDScriptParser::ClassNode *p_class = nullptr, const String &p_script_path = "") {
 	GDScriptParser::DataType type;
 	type.type_source = GDScriptParser::DataType::ANNOTATED_EXPLICIT;
 	type.kind = GDScriptParser::DataType::ENUM;
 	type.builtin_type = p_meta ? Variant::DICTIONARY : Variant::INT;
 	type.enum_type = p_enum_name;
 	type.is_constant = true;
 	type.is_meta_type = p_meta;
+	type.class_type = p_class;
```

Comment:
Given that we have make_global_enum type and make_native_enum_type which use this method as a base, I'd rather put adding a class into a new method make_class_enum_type and keep this method as the "common" logic

Reply:
Okay.. done, Check again